google-cloud-storage/src/main/java/com/google/cloud/storage/BucketInfo.java

```diff
@@ -184,17 +212,23 @@ static IamConfiguration fromPb(Bucket.IamConfiguration iamConfiguration) {
       Bucket.IamConfiguration.UniformBucketLevelAccess uniformBucketLevelAccess =
           iamConfiguration.getUniformBucketLevelAccess();
       DateTime lockedTime = uniformBucketLevelAccess.getLockedTime();
+      String publicAccessPrevention = iamConfiguration.getPublicAccessPrevention();
 
       return newBuilder()
           .setIsUniformBucketLevelAccessEnabled(uniformBucketLevelAccess.getEnabled())
           .setUniformBucketLevelAccessLockedTime(lockedTime == null ? null : lockedTime.getValue())
+          .setPublicAccessPrevention(
+              publicAccessPrevention == null
+                  ? null
+                  : PublicAccessPrevention.valueOf(publicAccessPrevention.toUpperCase()))
```

Comment:
`valueOf` isn't very safe for direct use without some pre validation as it will throw an `IllegalArgumentException` if the value passed in doesn't match.

Reply:
Apologies, missed this.

This would definitely suck. Do you have a suggestion?